### ISSUE TITLE
'subdirectory' config not supported for git reference in pip 

### ISSUE THREAD

**Issue by jday1** [allegroai/clearml#946]
Raised PR https://github.com/allegroai/clearml/pull/947 to solve this.

## Describe the bug

If I install a package from a github repository using the subdirectory feature, the url is not reflected correctly in ClearML's saved config.

E.g.
```
my_package @ git+https://github.com/my_owner/my_repository#egg=my_package&subdirectory=my_subdirectory
```
becomes
```
my_package @ git+https://github.com/my_owner/my_repository#egg=my_package
```

This causes an issue. For example, if I want to perform hyperparameter tuning for that job it will fail because the package installation directory is invalid. This can be solved by using the `detect_with_pip_freeze: true` flag. However, ideally this should work with Pigar as well. 

Read more about the `subdirectory` option here: https://packaging.python.org/en/latest/specifications/direct-url/#id1

## To reproduce
Install a package of that form such as:
`pip install 'airr @ git+https://github.com/SyntenyBio/airr-standards#egg=airr&subdirectory=lang/python'`

Run a clearml experiment with the package installed and `detect_with_pip_freeze: false` in config. 

In the clearml console under packages, you will see `airr` is malformed.

## Expected behaviour

Packages are saved as expected.

## Environment
* Server type (self hosted \ app.clear.ml)
* ClearML SDK Version
* ClearML Server Version (Only for self hosted). Can be found on the bottom right corner of the settings screen.
* Python Version
* OS (Windows \ Linux \ Macos)
## Related Discussion
If this continues a slack thread, please provide a link to the original slack thread.

**Comment by pollfly** [comment]
Hey @jday1 ! Just letting you know that this issue has been resolved in the recently released [v1.10.0](https://github.com/allegroai/clearml/releases/tag/v1.10.0). Let us know if there are any issues :)

**Comment by jkhenning** [comment]
@jday1 closing this, please reopen if required.
